### ISSUE TITLE
GuiFont! fails for bad monospace fonts

### ISSUE THREAD

**Issue by jgehrig** [equalsraf/neovim-qt#628]
@Mte90 is quoted below from #588:

Seems that with the latest master my ginit.vim is not executed.

My ginit.vim content is:
```
if exists('g:GuiLoaded')
    Guifont! DroidSansMono\ Nerd\ Font:h10
    GuiTabline 0
    call GuiWindowMaximized(1)
endif
```
I see that the font is not changed, if I run manually the font change is executed.
So on my testing I reverted some commits before the latest master and found if that the commit is 06b40dd if I set the commit before that the font is configured.

**Comment by jgehrig** [comment]
This is an interesting bug...

It occurs due to `m_nvim->api0()->vim_set_option("guifont", fontDesc());` in `setAttached`.

Here is the resulting call order:
```
guifont ""
setAttached
GuiFont "Liberation Sans:h14"
guifont "Liberation Mono:h11"
guifont "Liberation Sans:h14"
```

The call to `guifont "Liberation Mono:h11"` is a result of trying to update `guifont` so that it reflects the displayed font in `setAttached`. Since `set guifont` does not support an option for forcing bad monospace fonts, the default font (Liberation Mono) overrides the user-set font (Liberation Sans).

I think we can safely get rid of the problematic line... We will still update the value of `guifont` when `:GuiFont` is called. This behavior seems similar to other gui options.

**Comment by jgehrig** [comment]
Fixed in Pull #629, commit e2b0ac5d0736020573d125d094af0ba06c5bc362.
